source/adaptive_load/metrics_evaluator_impl.cc

```diff
@@ -15,16 +15,32 @@ using ::nighthawk::adaptive_load::MetricSpec;
 using ::nighthawk::adaptive_load::MetricSpecWithThreshold;
 using ::nighthawk::adaptive_load::ThresholdSpec;
 
+// Utility function for GetMetric functionality with fallback logic.
+absl::StatusOr<double> GetMetric(MetricsPlugin& metrics_plugin, absl::string_view metric_name,
+                                 const google::protobuf::Timestamp& start_time,
+                                 const google::protobuf::Duration& duration) {
+  absl::StatusOr<double> metric_value_or =
+      metrics_plugin.GetMetricByNameWithTime(metric_name, start_time, duration);
+  // If the metric plugin does not support WithTime implementation (i.e. is using the default
+  // implementation), default to GetMetricByName
+  if (metric_value_or.status().code() == absl::StatusCode::kUnimplemented) {
```

Comment:
Do we want to create a follow up task to eventually phase out the old interface?

Reply:
Yes please, could you file an issue describing what needs to be done?

Reply:
https://github.com/envoyproxy/nighthawk/issues/889